### PR TITLE
Refs #37102 - detect plugins webpack by looking for remoteEntry.js

### PR DESCRIPTION
**Pull request by evgeni**
Since we switched to Webpack 5, there are no more manifest.json in the plugins webpack folder, but we need some token to detect that we gotta load webpack stuff.

### CODE REVIEW

**Comment by evgeni** [comment]
For devel setups and Debian this works by checking `webpack/index.js`, but in RPM we exclude this folder from the installed GEM:
https://github.com/theforeman/foreman-packaging/blob/rpm/develop/gem2rpm/foreman_plugin.spec.erb#L30
(it's correct, we don't need the webpack stuff ON the target system, as we don't compile any assets there, but…)

File: app/registries/foreman/plugin/assets.rb [hunk 1]
```diff
@@ -30,8 +30,13 @@ def webpack_manifest_path
...
       end
 
+      def webpack_remote_entry
+        remote_entry_path = File.join(path, 'public', 'webpack', id.to_s, "#{id}_remoteEntry.js")
+        File.file?(remote_entry_path) ? remote_entry_path : nil
+      end
+
       def uses_webpack?
-        path && (File.file?(File.join(path, 'webpack', 'index.js')) || webpack_manifest_path.present?)
+        path && (File.file?(File.join(path, 'webpack', 'index.js')) || webpack_manifest_path.present? || webpack_remote_entry.present?)
```
Comment (evgeni) [inline app/registries/foreman/plugin/assets.rb:40]: arguably, `webpack_manifest_path.present?` can be dropped now
also, this method hits the disk on every page load, we should cache the result in memory (but scope creep!)

**Comment by evgeni** [comment]
With this applied I get working Katello prod installs with Webpack 5 \o/

File: app/registries/foreman/plugin/assets.rb (hunk 1 above)
Comment (ekohl) [reply app/registries/foreman/plugin/assets.rb:40]: Which is exactly what I aimed at in https://github.com/theforeman/foreman/pull/9834#discussion_r1329837249 but I should have pushed further on this.

Comment (evgeni) [reply app/registries/foreman/plugin/assets.rb:40]: you mean "hits the disk" part?

Comment (MariaAga) [inline app/registries/foreman/plugin/assets.rb:40]: When is ` path && (File.file?(File.join(path, 'webpack', 'index.js'))` false? I only see it as true all the time

Comment (evgeni) [reply app/registries/foreman/plugin/assets.rb:40]: when we use RPMs as we do not ship the `webpack` folder at all there, as we do not need to compile anything from it on the system where we deploy.

File: app/registries/foreman/plugin/assets.rb [hunk 2]
```diff
@@ -30,8 +30,13 @@ def webpack_manifest_path
         File.file?(manifest_path) ? manifest_path : nil
       end
 
+      def webpack_remote_entry
+        remote_entry_path = File.join(path, 'public', 'webpack', id.to_s, "#{id}_remoteEntry.js")
```
Comment (MariaAga) [inline app/registries/foreman/plugin/assets.rb]: probably should do the `-` -> `_` change for tasks here

Comment (evgeni) [reply app/registries/foreman/plugin/assets.rb]: Good catch! Updated

**Comment by evgeni** [comment]
[test unit]

seems unrelated

**Review by MariaAga — APPROVED** [review]
